src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesSignedCommits.java

```diff
@@ -0,0 +1,162 @@
+package com.sap.sgs.phosphor.fosstars.data.github;
+
+import static com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures.USES_VERIFIED_SIGNED_COMMITS;
+
+import com.sap.sgs.phosphor.fosstars.data.UserCallback;
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.ValueSet;
+import com.sap.sgs.phosphor.fosstars.model.value.BooleanValue;
+import com.sap.sgs.phosphor.fosstars.tool.YesNoSkipQuestion;
+import com.sap.sgs.phosphor.fosstars.tool.YesNoSkipQuestion.Answer;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Objects;
+import java.util.Optional;
+import org.json.JSONObject;
+import org.kohsuke.github.GHCommit;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * This data provider tries to figure out if a project uses signed commits. First, it checks if a
+ * project belongs to an organization and checks if any commits are signed or verified. It gathers
+ * the list of commits within one year from the current date. Then it calculates density ratio
+ * between number of signed commits vs unsigned commits. If the density ratio >= 1, then we set the
+ * flag = true else flag = false. Foundation. Next, it tries to ask a user if {@link UserCallback}
+ * is available.
+ */
+public class UsesSignedCommits extends AbstractGitHubDataProvider {
+
+  /**
+   * 1 year in millis.
+   */
+  private static final long DELTA = 365 * 24 * 60 * 60 * 1000L;
+
+  /**
+   * The percentage threshold of number of signed commits against number of total commits.
+   */
+  private static final double SIGNED_COMMIT_PRECENTAGE_THRESHOLD = 0.4;
+
+  /**
+   * A git token provided by user to access the Github API.
+   */
+  private final String gitToken;
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param where A GitHub organization of user name.
+   * @param name A name of a repository.
+   * @param github An interface to the GitHub API.
+   * @param gitToken provided by user to access GitHub API.
+   */
+  public UsesSignedCommits(String where, String name, GitHub github, String gitToken)
+      throws IOException {
+    super(where, name, github);
+    this.gitToken = gitToken;
+  }
+
+  @Override
+  public UsesSignedCommits update(ValueSet values) throws IOException {
+    Objects.requireNonNull(values, "Hey! Values can't be null!");
+    System.out.println("[+] Figuring out if the project uses any signed commits ...");
+
+    Optional<Value> something = cache().get(url, USES_VERIFIED_SIGNED_COMMITS);
+    if (something.isPresent()) {
+      values.update(something.get());
+      return this;
+    }
+
+    boolean flag = densityEvaluation();
+    if (callback.canTalk()) {
```

Comment:
I don't think we need to ask a user about this feature. This is pretty clear how the data provider can fill out the feature without a user.